╔══════════════════════════════════════════════════════════════════════════════╗
║                Welcome to Francinette, a 42 tester framework!                ║
╚═══════════════════════╦══════════════════════════════╦═══════════════════════╝
                        ║             libft            ║
                        ╚══════════════════════════════╝
✔ Preparing framework
✖ Executing: norminette
ft_atoi.c: OK!
ft_bzero.c: OK!
ft_calloc.c: OK!
ft_isalnum.c: OK!
ft_isalpha.c: OK!
ft_isascii.c: OK!
ft_isdigit.c: OK!
ft_isprint.c: OK!
ft_isspace.c: OK!
ft_itoa.c: OK!
ft_lstadd_back.c: OK!
ft_lstadd_front.c: OK!
ft_lstclear.c: OK!
ft_lstdelone.c: OK!
ft_lstiter.c: OK!
ft_lstlast.c: OK!
ft_lstmap.c: OK!
ft_lstnew.c: OK!
ft_lstsize.c: Error!
Error: WRONG_SCOPE_COMMENT  (line:  25, col:   5):	Comment is invalid in this scope
ft_memchr.c: OK!
ft_memcmp.c: OK!
ft_memcpy.c: OK!
ft_memmove.c: OK!
ft_memset.c: OK!
ft_putchar_fd.c: OK!
ft_putendl_fd.c: OK!
ft_putnbr_fd.c: OK!
ft_putstr_fd.c: OK!
ft_split.c: OK!
ft_strchr.c: OK!
ft_strdup.c: OK!
ft_striteri.c: OK!
ft_strjoin.c: OK!
ft_strlcat.c: OK!
ft_strlcpy.c: OK!
ft_strlen.c: OK!
ft_strmapi.c: OK!
ft_strncmp.c: OK!
ft_strnstr.c: OK!
ft_strrchr.c: OK!
ft_strrev.c: OK!
ft_strtrim.c: OK!
ft_substr.c: OK!
ft_tolower.c: OK!
ft_toupper.c: OK!
libft.h: OK!
✔ Executing: make fclean bonus
✔ Compiling tests: fsoares (my own)
ℹ Testing:
ft_lstmap       : KO
ft_split        : 1.KO Abort
Errors found:
For ft_lstmap, in /Users/fsoares-/fraaaaa/tests/libft/fsoares/test_lstmap.c:
Error in test 2: ft_lstmap(<list: {node: "hello!"}->(null)>, [s => __strlen(s)], [x => free(x)]) NULL check for 1th malloc:
Memory leak: 0x_address - 10 bytes
You failed to free the memory allocated at:
     in _add_malloc malloc_mock.c:29:8
     in malloc malloc_mock.c:75:10
  -> in map_length test_lstmap.c:6:8
     in ft_lstmap ft_lstmap.c:24:10
     in test_single_lstmap test_lstmap.c:41:17
     in test_lstmap test_lstmap.c:60:6
     in main test_lstmap.c:70:2
Most likely you are not calling del on the content when a new list node allocation fails.
Error in test 3: ft_lstmap(<list: {node: "one"}->{node: "two"}->{node: "three"}->{node: "four"}->{node: "five"}->(null)>, [s => __strlen(s)], [x => free(x)]) NULL check for 1th malloc:
Memory leak: 0x_address - 10 bytes
You failed to free the memory allocated at:
     in _add_malloc malloc_mock.c:29:8
     in malloc malloc_mock.c:75:10
  -> in map_length test_lstmap.c:6:8
     in ft_lstmap ft_lstmap.c:24:10
     in test_single_lstmap test_lstmap.c:41:17
     in test_lstmap test_lstmap.c:61:6
     in main test_lstmap.c:70:2
Most likely you are not calling del on the content when a new list node allocation fails.
Error in test 3: ft_lstmap(<list: {node: "one"}->{node: "two"}->{node: "three"}->{node: "four"}->{node: "five"}->(null)>, [s => __strlen(s)], [x => free(x)]) NULL check for 3th malloc:
Memory leak: 0x_address - 10 bytes
You failed to free the memory allocated at:
     in _add_malloc malloc_mock.c:29:8
     in malloc malloc_mock.c:75:10
  -> in map_length test_lstmap.c:6:8
     in ft_lstmap ft_lstmap.c:24:10
     in test_single_lstmap test_lstmap.c:41:17
     in test_lstmap test_lstmap.c:61:6
     in main test_lstmap.c:70:2
Most likely you are not calling del on the content when a new list node allocation fails.
Error in test 3: ft_lstmap(<list: {node: "one"}->{node: "two"}->{node: "three"}->{node: "four"}->{node: "five"}->(null)>, [s => __strlen(s)], [x => free(x)]) NULL check for 5th malloc:
Memory leak: 0x_address - 10 bytes
You failed to free the memory allocated at:
     in _add_malloc malloc_mock.c:29:8
     in malloc malloc_mock.c:75:10
  -> in map_length test_lstmap.c:6:8
     in ft_lstmap ft_lstmap.c:24:10
     in test_single_lstmap test_lstmap.c:41:17
     in test_lstmap test_lstmap.c:61:6
     in main test_lstmap.c:70:2
Most likely you are not calling del on the content when a new list node allocation fails.
...
File too large. To see full report open: /Users/fsoares-/fraaaaa/temp/libft/fsoares/error.log

=====> File /Users/fsoares-/fraaaaa/temp/libft/fsoares/error.log contents:
For ft_lstmap, in /Users/fsoares-/fraaaaa/tests/libft/fsoares/test_lstmap.c:

Error in test 2: ft_lstmap(<list: {node: "hello!"}->(null)>, [s => __strlen(s)], [x => free(x)]) NULL check for 1th malloc: 
Memory leak: 0x_address - 10 bytes
You failed to free the memory allocated at:
     in _add_malloc malloc_mock.c:29:8
     in malloc malloc_mock.c:75:10
  -> in map_length test_lstmap.c:6:8
     in ft_lstmap ft_lstmap.c:24:10
     in test_single_lstmap test_lstmap.c:41:17
     in test_lstmap test_lstmap.c:60:6
     in main test_lstmap.c:70:2
Most likely you are not calling del on the content when a new list node allocation fails.

Error in test 3: ft_lstmap(<list: {node: "one"}->{node: "two"}->{node: "three"}->{node: "four"}->{node: "five"}->(null)>, [s => __strlen(s)], [x => free(x)]) NULL check for 1th malloc: 
Memory leak: 0x_address - 10 bytes
You failed to free the memory allocated at:
     in _add_malloc malloc_mock.c:29:8
     in malloc malloc_mock.c:75:10
  -> in map_length test_lstmap.c:6:8
     in ft_lstmap ft_lstmap.c:24:10
     in test_single_lstmap test_lstmap.c:41:17
     in test_lstmap test_lstmap.c:61:6
     in main test_lstmap.c:70:2
Most likely you are not calling del on the content when a new list node allocation fails.

Error in test 3: ft_lstmap(<list: {node: "one"}->{node: "two"}->{node: "three"}->{node: "four"}->{node: "five"}->(null)>, [s => __strlen(s)], [x => free(x)]) NULL check for 3th malloc: 
Memory leak: 0x_address - 10 bytes
You failed to free the memory allocated at:
     in _add_malloc malloc_mock.c:29:8
     in malloc malloc_mock.c:75:10
  -> in map_length test_lstmap.c:6:8
     in ft_lstmap ft_lstmap.c:24:10
     in test_single_lstmap test_lstmap.c:41:17
     in test_lstmap test_lstmap.c:61:6
     in main test_lstmap.c:70:2
Most likely you are not calling del on the content when a new list node allocation fails.

Error in test 3: ft_lstmap(<list: {node: "one"}->{node: "two"}->{node: "three"}->{node: "four"}->{node: "five"}->(null)>, [s => __strlen(s)], [x => free(x)]) NULL check for 5th malloc: 
Memory leak: 0x_address - 10 bytes
You failed to free the memory allocated at:
     in _add_malloc malloc_mock.c:29:8
     in malloc malloc_mock.c:75:10
  -> in map_length test_lstmap.c:6:8
     in ft_lstmap ft_lstmap.c:24:10
     in test_single_lstmap test_lstmap.c:41:17
     in test_lstmap test_lstmap.c:61:6
     in main test_lstmap.c:70:2
Most likely you are not calling del on the content when a new list node allocation fails.

Error in test 3: ft_lstmap(<list: {node: "one"}->{node: "two"}->{node: "three"}->{node: "four"}->{node: "five"}->(null)>, [s => __strlen(s)], [x => free(x)]) NULL check for 7th malloc: 
Memory leak: 0x_address - 10 bytes
You failed to free the memory allocated at:
     in _add_malloc malloc_mock.c:29:8
     in malloc malloc_mock.c:75:10
  -> in map_length test_lstmap.c:6:8
     in ft_lstmap ft_lstmap.c:24:10
     in test_single_lstmap test_lstmap.c:41:17
     in test_lstmap test_lstmap.c:61:6
     in main test_lstmap.c:70:2
Most likely you are not calling del on the content when a new list node allocation fails.

Error in test 3: ft_lstmap(<list: {node: "one"}->{node: "two"}->{node: "three"}->{node: "four"}->{node: "five"}->(null)>, [s => __strlen(s)], [x => free(x)]) NULL check for 9th malloc: 
Memory leak: 0x_address - 10 bytes
You failed to free the memory allocated at:
     in _add_malloc malloc_mock.c:29:8
     in malloc malloc_mock.c:75:10
  -> in map_length test_lstmap.c:6:8
     in ft_lstmap ft_lstmap.c:24:10
     in test_single_lstmap test_lstmap.c:41:17
     in test_lstmap test_lstmap.c:61:6
     in main test_lstmap.c:70:2
Most likely you are not calling del on the content when a new list node allocation fails.


For ft_split, in /Users/fsoares-/fraaaaa/tests/libft/fsoares/test_split.c:

Error in test 1: ft_split("hello!", 32:' '): Memory problems!
     in sigabort utils.c:58:1
     in ft_strlcpy ft_strlcpy.c:25:20
     in ft_strlcpy ft_strlcpy.c:25:20
     in ft_substr ft_substr.c:35:10
     in populate ft_split.c:52:2
     in ft_split ft_split.c:88:8
     in test_single_split test_split.c:59:9
     in test_split test_split.c:79:6
     in main test_split.c:98:2

=================================================================
==pid==ERROR: AddressSanitizer: heap-buffer-overflow on address 0x_address at pc 0x_address bp 0x_address sp 0x_address
WRITE of size 1 at 0x_address thread T0
    #0 0x_address in ft_strlcpy ft_strlcpy.c:25
    #1 0x_address in ft_substr ft_substr.c:34
    #2 0x_address in populate ft_split.c:52
    #3 0x_address in ft_split ft_split.c:88
    #4 0x_address in test_single_split test_split.c:59
    #5 0x_address in test_split test_split.c:79
    #6 0x_address in main test_split.c:98
    #7 0x_address in start+0x0 (libdyld.dylib:x86_64+0x1acc8)

0x_address is located 0 bytes to the right of 6-byte region [0x_address,0x_address)
allocated by thread T0 here:
    #0 0x_address in wrap_malloc+0x9d (libclang_rt.asan_osx_dynamic.dylib:x86_64h+0x4917d)
    #1 0x_address in malloc malloc_mock.c:65
    #2 0x_address in ft_substr ft_substr.c:29
    #3 0x_address in populate ft_split.c:52
    #4 0x_address in ft_split ft_split.c:88
    #5 0x_address in test_single_split test_split.c:59
    #6 0x_address in test_split test_split.c:79
    #7 0x_address in main test_split.c:98
    #8 0x_address in start+0x0 (libdyld.dylib:x86_64+0x1acc8)

SUMMARY: AddressSanitizer: heap-buffer-overflow ft_strlcpy.c:25 in ft_strlcpy

